doc/dev_guide/internals/box_protocol.rst

```diff
@@ -1169,6 +1214,73 @@ To see how Tarantool handles this, look at
 `net_box.c <https://github.com/tarantool/tarantool/blob/master/src/box/lua/net_box.c>`_
 function ``netbox_encode_auth``.
 
+..  _box_protocol-streams:
+
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Binary protocol -- streams
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The streams feature, which was added in Tarantool version
+:tarantool-release:`2.10.0-beta1`, allows two things:
+sequential processing and interleaving.
+
+Sequential processing:
+With streams there is a guarantee that the server instance will not
+handle the next request in a stream until it has completed the previous one.
+
+Interleaving:
+A series of requests can include
+"insert for stream #1", "insert for stream #2", "delete for stream #2", "delete
+for stream #1", and then a "commit for stream #1" which the server instance will
+interpret as a request to handle only the two inserts for stream #1.
+
+To make these things possible,
+the client is responsible for ensuring that the stream identifier,
+unsigned integer :ref:`IPROTO_STREAM_ID <box_protocol-iproto_stream_id>`, is in the request header.
+IPROTO_STREAM_ID can be any positive 64-bit number, and should be unique for the connection.
+If IPROTO_STREAM_ID equals zero the server instance will ignore it.
+
+For example, suppose that the client has started a stream with
+the :ref:`net.box module <net_box-module>`
+
+..  code-block:: lua
+
+    net_box = require('net.box')
+    conn = net_box.connect('localhost:3302')
+    stream = conn:new_stream()
+
+At this point the stream object will look like a duplicate of
+the conn object, with just one additional member: ``stream_id``.
+Now, using stream instead of conn, the client sends two requests:
+
+..  code-block:: lua
+
+    stream.space.T:insert{1}
+    stream.space.T:insert{2}
+
+The header and body of these requests will be the same as in
+non-stream :ref:`IPROTO_INSERT <box_protocol-insert>` requests, except
+that the header will contain an additional item: IPROTO_STREAM_ID=0x0a
+with MP_UINT=0x01. It happens to equal 1 for this example because
+each call to conn:new_stream() assigns a new number, starting with 1.
+
+The client makes stream transactions by sending, in order:
+IPROTO_BEGIN, the transaction data-change and query requests,
+IPROTO_COMMIT or IPROTO_ROLLBACK.
+Each request must contain the same IPROTO_STREAM_ID value.
+With streaming there is no need to add
+:ref:`IPROTO_FLAGS <box_protocol-flags>` + IPROTO_FLAG_COMMIT
+in the header of the last request of a transaction.
+Rollback will be automatic if disconnect occurs before commit is possible.
+
+Thus there are now multiple ways to do transactions:
+with net_box and stream:begin() and stream:commit() or stream:rollback()
+which cause IPROTO_BEGIN and IPROTO_COMMIT or IPROTO_ROLLBACK with
```

Comment:
IPROTO_BEGIN used to start transaction, IRPOTO_COMMIT to commit transaction and IPROTO_ROLLBACK used to rollback transaction.
There are multiple ways to start transaction:
1. stream:begin()
2. stream:call('box.begin')
3. stream:eval('box.begin()')
4. stream:execute('START TRANSACTION') -- sql syntax
There are multiple ways to commit transactions:
1. stream:commit()
2. stream:call('box.commit')
3. stream:eval('box.commit()')
4. stream:execute('COMMIT')
There are multiple ways to rollback transactions:
1. stream:rollback()
2. stream:call('box.rollback')
3. stream:eval('box.rollback()')
4. stream:execute('ROLLBACK')

Reply:
Yes, but we won't need to mention all possibilities in the iproto description.

Reply:
Here I would write:
"Thus there are now multiple ways to do transactions with net_box: 
stream:begin() and stream:commit() or stream:rollback()
which cause IPROTO_BEGIN and IPROTO_COMMIT or IPROTO_ROLLBACK with ..."

Reply:
The current sentence has clauses delimited with semicolons
"with net_box and ....;"
"with box.begin() and ...;"
"with SQL and ...".
Changing the initial words to "with net_box:" would make
it appear that the whole sentence is about net_box, but
only the first clause is about net_box.
So I will leave this unchanged, but thanks for the suggestion.